packages/side-nav/src/SideNavItem.tsx

```diff
@@ -0,0 +1,285 @@
+    color: ${uiColors.blue.dark3};
+  }
+`;
+
+const activeTextStyle = css`
+  font-weight: bold;
+  color: ${uiColors.green.dark3};
+`;
+
+const disabledTextStyle = css`
+  color: ${uiColors.gray.light1};
+`;
+
+const linkStyle = css`
+  text-decoration: none;
+  color: inherit;
+`;
+
+interface SharedSideNavItemProps {
+  /**
+   * Whether or not the SideNavItem will be displayed as active.
+   */
+  active?: boolean;
+  /**
+   * Whether or not the SideNavItem will be displayed as disabled.
+   */
+  disabled?: boolean;
+  /**
+   * The value for aria-current if the SideNavItem is active
+   */
+  ariaCurrentValue?: AriaCurrentValue;
+  /**
+   * Class name that will be applied to the underlying MenuItem's root component.
+   */
+  className?: string;
+  /**
+   * Content that will appear inside of the underlying MenuItem's content wrapper.
+   */
+  children?: ReactNode;
+}
+
+interface LinkProps extends HTMLElementProps<'a'>, SharedSideNavItemProps {
+  href: string;
+}
+
+interface DivProps extends HTMLElementProps<'div'>, SharedSideNavItemProps {
+  href?: null;
+}
+
+type CustomElementProps = SharedSideNavItemProps & {
+  as: ElementType<any>;
+  [key: string]: any;
+};
+
+export type SideNavItemProps = LinkProps | DivProps | CustomElementProps;
+
+function usesCustomElement(
+  props: SideNavItemProps,
+): props is CustomElementProps {
+  return (props as any).as != null;
+}
+
+function usesLinkElement(props: LinkProps | DivProps): props is LinkProps {
+  return props.href != null;
+}
+
+const RootComponentTypes = {
+  Div: 'div',
+  Link: 'a',
+  Custom: 'as',
+} as const;
+
+/**
+ * # SideNavItem
+ *
+ * ```
+  <SideNavItem href="#homeward" active>
+    Link Title
+  </SideNavItem>
+ * ```
+ @param props.active Whether or not the SideNavItem will be displayed as active.
+ @param props.disabled Whether or not the SideNavItem will be displayed as disabled.
+ @param props.href When provided, the underlying MenuItem's root component will be rendered as an anchor with this href value.
+ @param props.ariaCurrentValue The value for aria-current if the SideNavItem is active
+ @param props.className Class name that will be applied to the underlying MenuItem's root component.
+ @param props.children Content that will appear inside of the underlying MenuItem's content wrapper.
+ *
+ */
+function SideNavItem(props: SideNavItemProps) {
+  const {
+    active = false,
+    disabled = false,
+    ariaCurrentValue = AriaCurrentValue.Page,
+    className,
+    children,
+  } = props;
+
+  const rest = omit(props, [
+    'as',
+    'active',
+    'disabled',
+    'ariaCurrentValue',
+    'className',
+    'children',
+  ]);
+
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+
+  let Root: ElementType<any>;
+
+  if (usesLinkElement(props)) {
+    Root = RootComponentTypes.Link;
+  } else if (usesCustomElement(props)) {
+    Root = props[RootComponentTypes.Custom];
+  } else {
+    Root = RootComponentTypes.Div;
```

Comment:
I think we should probably default to a button here. Since this is a navigation component, it's fair to assume that most, if not all, of the items should respond to a click.